vault/resource_pki_secret_backend_intermediate_cert_request_test.go

```diff
@@ -181,6 +181,122 @@ resource "vault_pki_secret_backend_intermediate_cert_request" "test" {
 `, path, signatureBits)
 }
 
+func TestPkiSecretBackendIntermediateCertRequest_key_usage(t *testing.T) {
+	path := "pki-" + strconv.Itoa(acctest.RandInt())
+
+	resourceName := "vault_pki_secret_backend_intermediate_cert_request.test"
+	testCheckFunc := []resource.TestCheckFunc{
+		resource.TestCheckResourceAttr(resourceName, "backend", path),
+		resource.TestCheckResourceAttr(resourceName, "type", "internal"),
+		resource.TestCheckResourceAttr(resourceName, "common_name", "test.my.domain"),
+		resource.TestCheckResourceAttr(resourceName, "uri_sans.#", "1"),
+		resource.TestCheckResourceAttr(resourceName, "uri_sans.0", "spiffe://test.my.domain"),
+	}
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: providerFactories,
+		PreCheck:          func() { testutil.TestAccPreCheck(t) },
+		CheckDestroy:      testCheckMountDestroyed("vault_mount", consts.MountTypePKI, consts.FieldPath),
+		Steps: []resource.TestStep{
+			{
+				Config: testPkiSecretBackendIntermediateCertRequestConfig_key_usage(path, ""),
+				Check:  resource.ComposeTestCheckFunc(append(testCheckFunc, resource.TestCheckNoResourceAttr(resourceName, consts.FieldKeyUsage))...),
+			},
+			{
+				Config: testPkiSecretBackendIntermediateCertRequestConfig_key_usage(path, `["certsign"]`),
+				Check: resource.ComposeTestCheckFunc(append(testCheckFunc,
+					resource.TestCheckResourceAttr(resourceName, consts.FieldKeyUsage+".#", "1"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldKeyUsage+".0", "certsign"))...),
+			},
+			{
+				Config: testPkiSecretBackendIntermediateCertRequestConfig_key_usage(path, `["keyagreement", "DecipherOnly"]`),
+				Check: resource.ComposeTestCheckFunc(append(testCheckFunc,
+					resource.TestCheckResourceAttr(resourceName, consts.FieldKeyUsage+".#", "2"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldKeyUsage+".0", "keyagreement"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldKeyUsage+".1", "DecipherOnly"))...),
+			},
+		},
+	})
+}
+
+func testPkiSecretBackendIntermediateCertRequestConfig_key_usage(path string, optionalKeyUsage string) string {
+	var signatureBits string
```

Comment:
Could we collapse all the various clones of this into a single function say `testPkiSecretBackendIntermediateCertRequestConfig_generic`, and have all these helper functions delegate the tweaked config to it? We have 3-4 copies of this terraform config now.

Reply:
Good idea, pushed a new changeset.